governator-core/src/main/java/com/netflix/governator/conditional/ConditionalBinder.java

```diff
@@ -0,0 +1,404 @@
+package com.netflix.governator.conditional;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.inject.Binder;
+import com.google.inject.Binding;
+import com.google.inject.ConfigurationException;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.Provider;
+import com.google.inject.Stage;
+import com.google.inject.TypeLiteral;
+import com.google.inject.binder.LinkedBindingBuilder;
+import com.google.inject.internal.Errors;
+import com.google.inject.spi.BindingTargetVisitor;
+import com.google.inject.spi.Dependency;
+import com.google.inject.spi.Message;
+import com.google.inject.spi.ProviderInstanceBinding;
+import com.google.inject.spi.ProviderWithExtensionVisitor;
+import com.google.inject.spi.Toolable;
+
+/**
+ * An API to bind multiple conditional candidates separately, only to evaluate and 
+ * resolve to one candidate when the type is actually injected.  
+ * ConditionalBinder is intended for use in a Module
+ * 
+ * <pre><code>
+ * public class SnacksModule extends AbstractModule {
+ *   protected void configure() {
+ *     ConditionalBinder&lt;Snack&gt; conditionalbinder
+ *         = ConditionalBinder.newConditionalBinder(binder(), Snack.class);
+ *     conditionalbinder.when(new ConditionalOnProperty("type", "twix")).toInstance(new Twix());
```

Comment:
Which form are you referring to?  Do you mean the long DSL?  This is consistent with the Guice DSL, especially multibinder.